src/xrpld/rpc/handlers/Simulate.cpp

```diff
@@ -0,0 +1,307 @@
+            return RPC::invalid_field_error("tx.Account");  // LCOV_EXCL_LINE
+        }
+
+        auto const srcAddressID =
+            parseBase58<AccountID>(tx_json[jss::Account].asString());
+        if (!srcAddressID.has_value())
+        {
+            return RPC::make_error(
+                rpcSRC_ACT_MALFORMED, RPC::invalid_field_message("tx.Account"));
+        }
+
+        std::shared_ptr<SLE const> sle =
+            context.app.openLedger().current()->read(
+                keylet::account(*srcAddressID));
+        if (!hasTicketSeq && !sle)
+        {
+            JLOG(context.app.journal("Simulate").debug())
+                << "Failed to find source account "
+                << "in current ledger: " << toBase58(*srcAddressID);
+
+            return rpcError(rpcSRC_ACT_NOT_FOUND);
+        }
+
+        tx_json[jss::Sequence] = hasTicketSeq
+            ? 0
+            : context.app.getTxQ().nextQueuableSeq(sle).value();
+    }
+
+    return std::nullopt;
+}
+
+// {
+//   tx_blob: <string> XOR tx_json: <object>,
+//   binary: <bool>
+// }
+Json::Value
+doSimulate(RPC::JsonContext& context)
+{
+    context.loadType = Resource::feeMediumBurdenRPC;
+
+    Json::Value jvResult;  // the returned result
+    Json::Value tx_json;   // the tx as a JSON
+
+    // check validity of `binary` param
+    if (context.params.isMember(jss::binary))
+    {
+        auto const binary = context.params[jss::binary];
+        if (!binary.isBool())
+        {
+            return RPC::invalid_field_error(jss::binary);
+        }
+    }
+
+    if (context.params.isMember(jss::tx_blob))
+    {
+        if (context.params.isMember(jss::tx_json))
+        {
+            // both `tx_blob` and `tx_json` included
+            return rpcError(rpcINVALID_PARAMS);
+        }
+        auto const blob = context.params[jss::tx_blob];
+        if (!blob.isString())
+        {
+            return RPC::invalid_field_error(jss::tx_blob);
+        }
+        auto unHexed = strUnHex(blob.asString());
+
+        if (!unHexed || !unHexed->size())
+            return RPC::invalid_field_error(jss::tx_blob);
+        try
+        {
+            SerialIter sitTrans(makeSlice(*unHexed));
+            tx_json = STObject(std::ref(sitTrans), sfGeneric)
+                          .getJson(JsonOptions::none);
+        }
+        catch (std::runtime_error& e)
+        {
+            return RPC::invalid_field_error(jss::tx_blob);
+        }
+    }
+    else if (context.params.isMember(jss::tx_json))
+    {
+        tx_json = context.params[jss::tx_json];
+        if (!tx_json.isObject())
+        {
+            return RPC::object_field_error(jss::tx_json);
+        }
+    }
+    else
+    {
+        // neither `tx_blob` nor `tx_json` included
+        return rpcError(rpcINVALID_PARAMS);
+    }
+
+    // basic sanity checks for transaction shape
+    if (!tx_json.isMember(jss::TransactionType))
+    {
+        return RPC::missing_field_error("tx.TransactionType");
+    }
```

Comment:
Please add new line

Reply:
fixed in 0bbf589f53a13069448d709faa5cea88b6ee547f